Content.Client/Sprite/SpriteFadeSystem.cs

```diff
@@ -46,46 +60,82 @@ private void OnFadingShutdown(EntityUid uid, FadingSpriteComponent component, Co
-            foreach (var ent in state.GetClickableEntities(mapPos))
-            {
-                if (ent == player ||
-                    !_fadeQuery.HasComponent(ent) ||
-                    !_spriteQuery.TryGetComponent(ent, out var sprite) ||
-                    sprite.DrawDepth < playerSprite.DrawDepth)
-                {
-                    continue;
-                }
-
-                if (!_fadingQuery.TryComp(ent, out var fading))
-                {
-                    fading = AddComp<FadingSpriteComponent>(ent);
-                    fading.OriginalAlpha = sprite.Color.A;
-                }
+            pointsOfInterest.Add((vp.PixelToMap(_inputManager.MouseScreenPosition.Position), true));
+        }
 
-                _comps.Add(fading);
-                var newColor = Math.Max(sprite.Color.A - change, TargetAlpha);
+        if (TryComp(player, out TransformComponent? playerXform))
+        {
+            pointsOfInterest.Add((_transform.GetMapCoordinates(_playerManager.LocalEntity!.Value, xform: playerXform), false));
+        }
 
-                if (!sprite.Color.A.Equals(newColor))
+        if (_stateManager.CurrentState is GameplayState state && _spriteQuery.TryGetComponent(player, out var playerSprite))
+        {
+            foreach (var (mapPos, excludeBB) in pointsOfInterest)
+            {
+                // Also want to handle large entities even if they may not be clickable.
+                foreach (var ent in state.GetClickableEntities(mapPos, excludeFaded: false))
                 {
-                    sprite.Color = sprite.Color.WithAlpha(newColor);
+                    if (ent == player ||
+                        !_fadeQuery.HasComponent(ent) ||
+                        !_spriteQuery.TryGetComponent(ent, out var sprite) ||
+                        sprite.DrawDepth < playerSprite.DrawDepth)
+                    {
+                        continue;
+                    }
+
+                    if (excludeBB)
+                    {
+                        var test = new Box2Rotated(mapPos.Position - MouseRadius, mapPos.Position + MouseRadius);
+                        var collided = false;
+                        foreach (var fixture in _physics.GetCollidingEntities(mapPos.MapId, test))
+                        {
+                            if (fixture.Owner == ent)
+                            {
+                                collided = true;
+                                break;
+                            }
+                        }
+                        if (collided)
+                        {
+                            break;
+                        }
+                    }
+
+                    if (!_fadingQuery.TryComp(ent, out var fading))
+                    {
+                        fading = AddComp<FadingSpriteComponent>(ent);
+                        fading.OriginalAlpha = sprite.Color.A;
+                    }
```

Comment:
This is just ensurecomp

Reply:
not my code, this is just moved from preexisting